APIComparer.Backend.Tests/TopXNugets.cs

```diff
@@ -0,0 +1,212 @@
+﻿namespace APIComparer.Backend.Tests
+{
+    using NuGet;
+    using NUnit.Framework;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Reporting;
+    using VersionComparisons;
+
+    [TestFixture]
+    public class TopXNugets
+    {
+        IPackageRepository repo = PackageRepositoryFactory.Default.CreateRepository("https://packages.nuget.org/api/v2");
+
+        internal static string[] Frameworks =
+        {
+            ".NETFramework,Version=v4.5",
+            ".NETFramework,Version=v4.0"
+        };
+
+
+        [Test]
+        public void Should_calculate_stats_for_top_25_packages()
+        {
+            var packages = repo.Search(String.Empty, Frameworks, false)
+                .Skip(0)
+                .Take(50)
+                .ToList();
+
+            var statCollector = new StatsCollector(repo, packages);
+            statCollector.Limit = 25;
+
+            statCollector.Report();
+        }
+
+        [Test]
+        public void Should_calculate_stats_for_NServiceBus_packages()
+        {
+            var packages = repo.Search("NServiceBus", Frameworks, false)
+                .Skip(0)
+                .Take(50)
+                .ToList()
+                .Where(p => p.Authors.Contains("NServiceBus Ltd"))
+                .OrderByDescending(p => p.DownloadCount)
+                .ToList();
+
+            var statCollector = new StatsCollector(repo, packages);
+            statCollector.Filter = "^NServiceBus";
+            statCollector.SkipPackages = new[]
+            {
+                "NServiceBus.AcceptanceTesting",
+                "NServiceBus.RavenDB",
+                "NServiceBus.Wire"
+            };
+
+            statCollector.Report();
+        }
+
+        public class StatsCollector
```

Comment:
Started out as a private method but it was accumulating too much state. It's tightly coupled to the two "tests".